_posts/2020-06-11-Solidity-0610-release-announcement.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.6.10 Release Announcement'
+date: '2020-06-11'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.6.10](https://github.com/ethereum/solidity/releases/tag/v0.6.10) fixes an important bug that was introduced in the previous release and adds error codes.
+
+Please upgrade if you are using internal library functions with calldata paramaters in connection with ``using for``.
+
+## Important Bugfixes
+
+### Invalid Access for Calldata Parameters in Internal Library Functions
+
+**Problem**: The previous release, [Solidity 0.6.9](https://github.com/ethereum/solidity/releases/tag/v0.6.9), introduced the possibility to use ``calldata`` types in internal and not, as previously, only in external functions.
+This resulted in a bug related to calling library functions with calldata parameters via ``using for``.
+More specifically, the compiler would copy all calldata arguments to memory first - like done for calls to public library functions.
+The internal library function would then receive the memory pointer but interpret it as a calldata pointer resulting in
+calldata reads at the wrong location.
```

Comment:
```suggestion
calldata reads at the wrong location and potential stack corruption. The stack corruption is caused by the fact that calldata pointers can use two stack slots while memory pointers always only use one stack slot. This can in turn cause problems for other variables and a wrong jump at the function return.
```